include/aos/common/tools/list.hpp

```diff
@@ -0,0 +1,616 @@
+         *
+         * @return Iterator&.
+         */
+        Iterator operator++(int)
+        {
+            Iterator it(*this);
+
+            operator++();
+
+            return it;
+        }
+
+        /**
+         * Prefix decrement operator.
+         *
+         * @return Iterator&.
+         */
+        Iterator& operator--()
+        {
+            mCurrentNode = mCurrentNode->mPrev;
+
+            return *this;
+        }
+
+        /**
+         * Postfix decrement operator.
+         *
+         * @return Iterator&.
+         */
+        Iterator operator--(int)
+        {
+            Iterator it(*this);
+
+            operator--();
+
+            return it;
+        }
+
+        /**
+         * Checks if iterators are not equal.
+         *
+         * @param other iterator to compare with.
+         * @return bool.
+         */
+        bool operator!=(const Iterator& other) const { return mCurrentNode != other.mCurrentNode; }
+
+    private:
+        friend class List<T>;
+
+        Node* mCurrentNode = nullptr;
+    };
+
+    /**
+     * List const iterator.
+     */
+    class ConstIterator {
+    public:
+        /**
+         *  Creates const iterator.
+         */
+        ConstIterator() = default;
+
+        /**
+         *  Creates const iterator.
+         *
+         * @param node list node.
+         */
+        explicit ConstIterator(const Node* node)
+            : mCurrentNode(const_cast<Node*>(node))
+        {
+        }
+
+        /**
+         * Dereference operator.
+         *
+         * @return T&.
+         */
+        const T& operator*() const { return mCurrentNode->mValue; }
+
+        /**
+         * Prefix increment operator.
+         *
+         * @return ConstIterator&.
+         */
+        ConstIterator& operator++()
+        {
+            mCurrentNode = mCurrentNode->mNext;
+
+            return *this;
+        }
+
+        /**
+         * Postfix increment operator.
+         *
+         * @return ConstIterator&.
+         */
+        ConstIterator operator++(int)
+        {
+            ConstIterator it(*this);
+
+            operator++();
+
+            return it;
+        }
+
+        /**
+         * Prefix decrement operator.
+         *
+         * @return ConstIterator&.
+         */
+        ConstIterator& operator--()
+        {
+            mCurrentNode = mCurrentNode->mPrev;
+
+            return *this;
+        }
+
+        /**
+         * Postfix decrement operator.
+         *
+         * @return ConstIterator&.
+         */
+        ConstIterator operator--(int)
+        {
+            ConstIterator it(*this);
+
+            operator--();
+
+            return it;
+        }
+
+        /**
+         * Checks if iterators are not equal.
+         *
+         * @param other iterator to compare with.
+         * @return bool.
+         */
+        bool operator!=(const ConstIterator& other) const { return mCurrentNode != other.mCurrentNode; }
```

Comment:
operator ==  should be added for consistency

Reply:
done